src/Api/include/Api/EncodedString.h

```diff
@@ -0,0 +1,37 @@
+// Copyright (c) 2021 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef ORBIT_API_ENCODED_STRING_H_
+#define ORBIT_API_ENCODED_STRING_H_
+
+#include <cstddef>
+#include <cstdint>
+#include <string>
+
+// In order to avoid expensive allocations in proto buffers, we encode the first 64 characters in
+// eight fixed64 fields (byte by byte). Any additional characters will be also encoded in a sequence
+// of 64bit fields. So a string [0x11, 0x22, 0x33] would get encoded as 0x1122330000000000, all
+// other fields are zero and there are no additional fields.
```

Comment:
It confuses me quite a bit that the first byte of the string is encoded in the highest byte of the `uint64_t`. That is basically inverting the order of the bytes. This also prevents `memcpy`s because it goes against the endianness.